src/hydra/dataProvider.ts

```diff
@@ -281,10 +286,17 @@ function dataProvider(
       ).forEach(([key, value]) => {
         // React-Admin FileInput format is an object containing a file.
         if (containFile(value)) {
-          body.append(
-            key,
-            Object.values(value).find((val) => val instanceof File),
-          );
+          const findFile = (element: string | ToJSONObject): Blob =>
+            Object.values(element).find((val) => val instanceof File);
+          // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+          Array.isArray(value)
+            ? value
+                .map((val) => findFile(val))
+                .forEach((file) => {
+                  body.append(key.endsWith('[]') ? key : `${key}[]`, file);
```

Comment:
Is the ternary condition needed?
Could the key not end with `[]`?

Reply:
Just to be meticulous, not harmful but can handle some corner cases, and the question is rather opposite - "could the key end with `[]`?".  Very seldom, only  when set by explicit.  To support nicely and transparently all resource labels, its translations,  etc.,  key should by passed without `[]` suffix to all InputGuesser/ InputField components, so it is done this way right now, and that why we need to transform it here adding this array `[]` suffix.